### ISSUE TITLE
Standardising Numpy arrays - user-defined function output differs from repeated preprocessing logic

### ISSUE THREAD

**Issue by Data2Info2Knowledge** [khuyentran1401/production-ready-data-science-code#4]
In the /chapter5_functions.ipynb notebook, under "Why Are Python Functions Essential?", the two code cells give different results.
Specifically, in the second code snippet, each of the training and test sets is standardised independently based on its own mean and standard deviation, unlike best practice.
I'd suggest using a modified function that returns not only the standardised array but also the mean and standard deviation, which can then be reused (through optional arguments) for the test set, as in the example below.
```
def standardize_features(array, mean=None, std=None):
    if mean is None:
        mean = np.mean(array, axis=0)
    if std is None:
        std = np.std(array, axis=0)
    return (array - mean) / std, mean, std

X_train_standardized, train_mean, train_std = standardize_features(X_train)
X_test_standardized, _, _ = standardize_features(X_test, mean=train_mean, std=train_std)
```
I realise that in a real-world project we'd probably use scikit-learn's StandardScaler, but this is to make the illustration consistent.
